src/transformers/pipelines/video_classification.py

```diff
@@ -0,0 +1,113 @@
+
+    from decord import VideoReader, cpu
+
+
+if is_torch_available():
+    from ..models.auto.modeling_auto import MODEL_FOR_VIDEO_CLASSIFICATION_MAPPING
+
+logger = logging.get_logger(__name__)
+
+
+@add_end_docstrings(PIPELINE_INIT_ARGS)
+class VideoClassificationPipeline(Pipeline):
+    """
+    Video classification pipeline using any `AutoModelForVideoClassification`. This pipeline predicts the class of a
+    video.
+
+    This video classification pipeline can currently be loaded from [`pipeline`] using the following task identifier:
+    `"video-classification"`.
+
+    See the list of available models on
+    [huggingface.co/models](https://huggingface.co/models?filter=video-classification).
+    """
+
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+        requires_backends(self, "decord")
+        self.check_model_type(MODEL_FOR_VIDEO_CLASSIFICATION_MAPPING)
+
+        self.frame_sampling_rate = kwargs.pop("frame_sample_rate", 4)
+        self.num_frames = self.model.config.num_frames
+
+    def _sanitize_parameters(self, top_k=None):
+        postprocess_params = {}
+        if top_k is not None:
+            postprocess_params["top_k"] = top_k
+        return {}, {}, postprocess_params
+
+    def __call__(self, videos: Union[str, List[str]], **kwargs):
+        """
+        Assign labels to the video(s) passed as inputs.
+
+        Args:
+            videos (`str`, `List[str]`):
+                The pipeline handles three types of videos:
+
+                - A string containing a http link pointing to a video
+                - A string containing a local path to an video
+
+                The pipeline accepts either a single video or a batch of videos, which must then be passed as a string.
+                Videos in a batch must all be in the same format: all as http links or all as local paths.
+            top_k (`int`, *optional*, defaults to 5):
+                The number of top labels that will be returned by the pipeline. If the provided number is higher than
+                the number of labels available in the model configuration, it will default to the number of labels.
+
+        Return:
+            A dictionary or a list of dictionaries containing result. If the input is a single video, will return a
+            dictionary, if the input is a list of several videos, will return a list of dictionaries corresponding to
+            the videos.
+
+            The dictionaries contain the following keys:
+
+            - **label** (`str`) -- The label identified by the model.
+            - **score** (`int`) -- The score attributed by the model for that label.
+        """
+        return super().__call__(videos, **kwargs)
+
+    def preprocess(self, video):
+
+        if video.startswith("http://") or video.startswith("https://"):
+            video = BytesIO(requests.get(video).content)
+
+        videoreader = VideoReader(video, num_threads=1, ctx=cpu(0))
```

Comment:
What is the `VideoReader` object ?

I don't feel good about using stuff like threading of `cpu(0)` as `preprocess` is already supposed to be on specific threads (with `DataLoader` for PyTorch). 

If there is a simple single threaded load variant I think it would fit better.

If video loading uses all threads by default and it makes more sense that way we might want to handle the case where the pipeline is suposed to start multithreading to deactivate it.

Reply:
Actually went to check the dependency and I'm not sure it's worthwile, if it's using `ffmpeg` under the hood, we can write a simple function that will do what's required using subprocess quite easily.

Reply:
The problem with ffmpeg + subprocess is that it will fail silently, no?

Reply:
Depends on the code, mostly no. At least it shouldn't.